HLTrigger/special/plugins/HLTPPSJetComparisonFilter.cc

```diff
@@ -39,6 +41,10 @@ class HLTPPSJetComparisonFilter : public edm::global::EDFilter<> {
   const edm::ESGetToken<LHCInfo, LHCInfoRcd> lhcInfoToken_;
   const edm::ESGetToken<LHCInfoPerLS, LHCInfoPerLSRcd> lhcInfoPerLSToken_;
   const edm::ESGetToken<LHCInfoPerFill, LHCInfoPerFillRcd> lhcInfoPerFillToken_;
+
+  mutable edm::ESWatcher<LHCInfoPerLSRcd> lhcInfoLSWatcher_;
+  mutable edm::ESWatcher<LHCInfoPerFillRcd> lhcInfoFillWatcher_;
```

Comment:
as this is an `edm::global::EDFilter` this needs to be `mutable`. I am not sure if there is a better solution frame-work wise (e.g. by using the global cache).

Reply:
The `edm::ESWatcher` is not thread safe, so it can't be used in an `edm::global` module like this. We need to think a bit what to suggest instead.

Reply:
> , it looks like the code just wants to avoid printing excessive numbers of messages?

yes that's correct. Ideally we'd like to print once per LS.

>  We need to think a bit what to suggest instead.

Thanks!

Reply:
If that is true, since conditions can only change on LuminosityBlock boundaries, why not just do the print unconditionally on calls to globalBeginLuminosityBlock?